### DISCUSSION TITLE
Rank Mirrors addon takes excessive time to complete

### DISCUSSION THREAD

**Discussion by Sagnasty** [rern/rAudio#404] [Fixed Issues]
I did some more testing today on this today.

From the au to za mirrors it takes less than 5 seconds for each result to be displayed.

Next, the process then takes **5 1/2 minutes** to post the tw mirror results (although the tw reported time does not reflect this).  The ranking process does not hang as I previously indicated in bug report (https://github.com/rern/rAudio-1/discussions/379) by @R101-Github 

Lastly, the ranking process continues to post the remaining mirror results (us mirrors - less than 5 seconds for each result to be displayed) and then completes the ranking process.

Good news is that the behavior is repeatable.

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

![mirror](https://user-images.githubusercontent.com/76575081/122435445-01dff380-cf66-11eb-8e96-5713efc2c0a1.jpg)

**Comment by R101G** [comment]
I am now getting similar results in the UK. The error message is gone, and the final result is correct, but it takes about 5 minutes to complete. Most of the time is spent waiting for tw.

**Comment by Sagnasty** [comment]
I tried this again today after the improvements were added.   

Most times it works great, but sometimes I receive the following error below if the mirror ranking addon is run numerous times (see screenshot).

I have been able to repeat this error on two different pi's (B+ Rev 1.3 and 3B Rev 1.2).

Jim

3 x Raspberry Pi 3B+ Rev 1.3 > HiFiBerry DAC+ Pro
1 x Raspberry Pi 3B Rev 1.2 > HiFiBerry DAC+ Pro
Synology DS718+

![mirror error](https://user-images.githubusercontent.com/76575081/122603235-0b3e8e00-d042-11eb-9ea0-1f36dabdeaac.jpg)

**Comment by Sagnasty** [comment]
Tried 10 times today without error and consistent ranking results.

Job well done Rern!
